framework/src/modules/interoperability/sidechain/commands/mainchain_registration.ts

```diff
@@ -129,19 +146,32 @@ export class MainchainRegistrationCommand extends BaseInteroperabilityCommand {
 			ownName,
 			mainchainValidators,
 		});
-		bls.verifyWeightedAggSig(
-			keyList,
-			aggregationBits,
-			signature,
-			TAG_CHAIN_REG_MESSAGE,
-			chainID,
-			message,
-			weights,
-			certificateThreshold,
-		);
+
+		if (
+			!bls.verifyWeightedAggSig(
+				keyList,
+				aggregationBits,
+				signature,
+				TAG_CHAIN_REG_MESSAGE,
+				ownChainID,
+				message,
+				weights,
+				certificateThreshold,
+			)
+		) {
+			// emit persistent event with empty data
+			context.eventQueue.add(
```

Comment:
Event should be created under `events` instead of directly using `eventQueue`

Reply:
I think it's not possible with empty data field ?

Reply:
ah `data` is empty here. can you check with research team if it's correct to have empty data event?
If yes, then we should update BaseEvent to allow empty data event

Reply:
yes, it was done as per the suggestion from research team

Reply:
I created a new issue to update `BaseEvent`: https://github.com/LiskHQ/lisk-sdk/issues/7546